File: sensor-data-java/src/main/java/pipelines/examples/sensordata/SensorDataStreamingIngress.java
package pipelines.examples.sensordata;

import akka.http.javadsl.common.EntityStreamingSupport;
import akka.http.javadsl.marshallers.jackson.Jackson;

import pipelines.akkastream.AkkaServerStreamlet;

import pipelines.akkastream.util.javadsl.HttpServerLogic;
import pipelines.akkastream.StreamletLogic;
import pipelines.streamlets.RoundRobinPartitioner;
import pipelines.streamlets.StreamletShape;
import pipelines.streamlets.avro.AvroOutlet;

public class SensorDataStreamingIngress extends AkkaServerStreamlet {

  AvroOutlet<SensorData> out =  AvroOutlet.<SensorData>create("out", SensorData.class)
          .withPartitioner(RoundRobinPartitioner.getInstance());

  public StreamletShape shape() {
   return StreamletShape.createWithOutlets(out);
  }

  public StreamletLogic createLogic() {
    EntityStreamingSupport ess = EntityStreamingSupport.json();
    return HttpServerLogic.createDefaultStreaming(this, out, Jackson.byteStringUnmarshaller(SensorData.class), ess, getStreamletContext());
  }
}

File: sensor-data-scala/src/main/scala/pipelines/examples/sensordata/SensorDataStreamingIngress.scala
package pipelines.examples.sensordata

import akka.http.scaladsl.common.EntityStreamingSupport
import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._

import SensorDataJsonSupport._
import pipelines.akkastream.AkkaServerStreamlet
import pipelines.akkastream.util.scaladsl._
import pipelines.streamlets.{ RoundRobinPartitioner, StreamletShape }
import pipelines.streamlets.avro._

class SensorDataStreamingIngress extends AkkaServerStreamlet {
  val out = AvroOutlet[SensorData]("out", RoundRobinPartitioner)
  def shape = StreamletShape.withOutlets(out)

  implicit val entityStreamingSupport = EntityStreamingSupport.json()
  override def createLogic = HttpServerLogic.defaultStreaming(this, out)
}
